### PR TITLE
FIREFLY-723: adding a pre message above Title for SOFIA/Spitzer need

### PR DESCRIPTION
**Pull request by ejoliet**
This PR is a change to add a message above the Title of the download dialog.

This is needed for SOFIA/Spitzer who wants to add some warning depending of the table/file to be downloaded to give the user more info about what they are downloading (IRSA-3944).

No build for testing here, see an example to test in IRSA-3944 PR: https://github.com/IPAC-SW/irsa-ife/pull/154

### CODE REVIEW

File: src/firefly/js/ui/DownloadDialog.jsx [hunk 1]
```diff
@@ -185,7 +191,10 @@ export function DownloadOptionPanel (props) {
     };
 
     const dlTitle = get(dlParams, 'TitlePrefix', 'Download') + '-' + dlTitleIdx;
-
+    // IRSA-3944: SOFIA required a message to guide the user before downloading AOR such as:
+    // 'All levels associated with this AOR will be downloaded. To download individual data products, go to the relevant instrument tab.'
+    // PreTitleMessage can be added to the dlParams from the parent and passed here (FIREFLY-723)
+    const preTitleMessage = get(dlParams, 'PreTitleMessage', '')
```
Comment (robyww) [inline src/firefly/js/ui/DownloadDialog.jsx:197]: we don't need `get` anymore
```
const preTitleMessage = dlParams?.PreTitleMessage ?? '';
```

File: src/firefly/js/ui/DownloadDialog.jsx [hunk 2]
```diff
@@ -185,7 +191,10 @@ export function DownloadOptionPanel (props) {
     };
 
     const dlTitle = get(dlParams, 'TitlePrefix', 'Download') + '-' + dlTitleIdx;
-
+    // IRSA-3944: SOFIA required a message to guide the user before downloading AOR such as:
+    // 'All levels associated with this AOR will be downloaded. To download individual data products, go to the relevant instrument tab.'
+    // PreTitleMessage can be added to the dlParams from the parent and passed here (FIREFLY-723)
```
Comment (robyww) [inline src/firefly/js/ui/DownloadDialog.jsx:196]: This message should probably be at the top with the other documentation

**Review by robyww — APPROVED** [review]
I mentioned a couple of small things to clean up. Then it is good to merge.

File: src/firefly/js/ui/DownloadDialog.jsx (hunk 1 above)
Comment (ejoliet) [reply src/firefly/js/ui/DownloadDialog.jsx:197]: cool, i'll change it before merging.

File: src/firefly/js/ui/DownloadDialog.jsx (hunk 2 above)
Comment (ejoliet) [reply src/firefly/js/ui/DownloadDialog.jsx:196]: good, i'll update it.

File: src/firefly/js/ui/DownloadDialog.jsx (hunk 1 above)
Comment (ejoliet) [reply src/firefly/js/ui/DownloadDialog.jsx:197]: @robyww are you sure? It says unresolved variable - are you sure of the syntax?

Comment (robyww) [reply src/firefly/js/ui/DownloadDialog.jsx:197]: yes, use it all the time

Comment (ejoliet) [reply src/firefly/js/ui/DownloadDialog.jsx:197]: Ok, i'll copy paste and merge.

Comment (ejoliet) [reply src/firefly/js/ui/DownloadDialog.jsx]: Yep, confirmed. Seems intellij is not good enough to interpret the syntax.

**Pull request merged**